api/src/routers/meals.js

```diff
@@ -67,8 +67,117 @@ mealsRouter.get("/last-meal", async (req, res) => {
 // Get /meals
 mealsRouter.get("/meals", async (req, res) => {
   try {
-    const meals = await db("meals").select("*");
-    res.json(meals);
+    if ("maxPrice" in req.query) {
+      const maxPrice = req.query.maxPrice;
+
+      if (!isNaN(maxPrice) && maxPrice > 0) {
+        const limitedPriceMeals = await db("meals")
+          .select("*")
+          .where("price", "<", maxPrice);
+        res.json(limitedPriceMeals);
+        return;
+      }
+    } else if ("availableReservations" in req.query) {
+      const availableReservations =
+        req.query.availableReservations.toLowerCase();
+
+      if (availableReservations === "true") {
+        const availableReservationsMeals = await db("meals")
+          .leftJoin("reservations", "meals.id", "reservations.meal_id")
+          .groupBy("meals.id")
+          .select("meals.*")
+          .havingRaw("meals.max_reservations > COUNT(reservations.id)");
+
+        res.json(availableReservationsMeals);
+        return;
+      } else if (availableReservations === "false") {
+        const unAvailableReservationsMeals = await db("meals")
+          .leftJoin("reservations", "meals.id", "reservations.meal_id")
+          .groupBy("meals.id")
+          .select("meals.*")
+          .havingRaw("meals.max_reservations <= COUNT(reservations.id)");
+
+        res.json(unAvailableReservationsMeals);
+        return;
+      }
+    } else if ("title" in req.query) {
+      const title = req.query.title.toString().toLowerCase();
+
+      if (title) {
+        const meals = await db("meals")
+          .select("*")
+          .where("title", "like", `%${title}%`);
+
+        res.json(meals);
+        return;
+      }
+    } else if ("dateAfter" in req.query) {
+      const dateAfter = new Date(req.query.dateAfter.toString());
+
+      if (dateAfter == "Invalid Date") {

```

Comment:
can you explain this check here? are you parsing dateAfter somewhere in the code?